apps/system/js/tracking_notice.js

```diff
@@ -0,0 +1,152 @@
+    this.options = {};
+    this._settings = navigator.mozSettings;
+    var req = navigator.mozSettings.createLock().get(TRACKING_NOTICE_KEY);
+    req.onsuccess = (function() {
+      var alreadyShown = req.result[TRACKING_NOTICE_KEY];
+      this._alreadyShown = alreadyShown;
+      if (!alreadyShown) {
+        this.renderDialog();
+      }
+    }.bind(this));
+
+    req.onerror = (function() {
+      this._alreadyShown = false;
+      this.renderDialog();
+    }.bind(this));
+
+    this.publish('created');
+  };
+
+  TrackingNotice.prototype = Object.create(SystemDialog.prototype);
+
+  TrackingNotice.prototype.customID = 'tracking-notice';
+
+  TrackingNotice.prototype.DEBUG = false;
+
+  TrackingNotice.prototype.name = 'TrackingNotice';
+
+  TrackingNotice.prototype.containerElement =
+    document.getElementById('fullscreen-overlay');
+
+  TrackingNotice.prototype.renderDialog = function() {
+    LazyLoader.load('shared/elements/gaia_switch/script.js', function() {
+      this.render();
+      this.boundShowIfNeeded = this.showIfNeeded.bind(this);
+      window.addEventListener('appopened', this.boundShowIfNeeded);
+    }.bind(this));
+  };
+
+  TrackingNotice.prototype.view = function view() {
+    return `<section id="tracking-notice" data-z-index-level="system-dialog"
+              class='hidden dialog'>
+              <div class="panel">
+                <header><h1 data-l10n-id="tracking-notice-header"></h1></header>
+                <p id="tracking-notice-copy"
+                data-l10n-id="tracking-notice-description"></p>
+                <img id="tracking-notice-image"></img>
+                <span data-l10n-id="tracking-protection"></span>
+                <div class="setting">
+                  <label data-l10n-id="tracking-protection"
+                  for="tracking-protection-toggle"></label>
+                  <gaia-switch id="tracking-protection-toggle">
+                  </gaia-switch>
+                  <p data-l10n-id="tracking-protection-setting-description"
+                  class="subtitle"></p>
+                </div>
+              </div>
+              <div class="confirm" id="tracking-notice-confirm">
+                <p data-l10n-id="tracking-notice-confirm"></p>
+              </div>
+            </section>`;
+  };
+
+
+  TrackingNotice.prototype._registerEvents = function() {
+    var self = this;
+    self.confirm.addEventListener('click', function onConfirm() {
+      self.confirm.removeEventListener('click', onConfirm);
+      self.confirmNotice();
+    });
+  };
+
+  TrackingNotice.prototype._fetchElements = function spl_initElements() {
+    this.element = document.querySelector('#' + this.instanceID);
+    this.confirm = this.element.querySelector('#tracking-notice-confirm');
+    this.setting = this.element.querySelector('gaia-switch');
+    this.description = this.element.querySelector('#tracking-notice-copy');
+    this.description.addEventListener('click', function() {
```

Comment:
It's pretty surprising to click anywhere on the description and have it navigate the underlying browser window. The biggest change I want here is to make only clicking on the link perform the action. Secondly, I would prefer that it would open a new window rather than navigate my already open one. But that second one I won't block on.